### ISSUE TITLE
[Bug] random mapping not working on pokemon omega ruby

### ISSUE THREAD

**Issue by Tommy-TI** [Ajarmar/universal-pokemon-randomizer-zx#386]
**Describe the bug**
The wild pokemon random selection and area 1-to-1 mapping generate the mapping of pokemon in the log.

**To Reproduce**
1. Go to 'Open ROM'
2. Click on 'Wild Pokemon'
3. Select 'Random'
4. Go check the log
5. it doesn't have a random mapping but a area 1-to-1

**Expected behavior**
I wanted the log to not contains duplicate of the same pokemon in each region.

**Screenshots**
here's a screenshot of a log generated in random mapping
![image](https://user-images.githubusercontent.com/58239919/152085585-14a681d4-459c-47e0-8764-80f31d4fee6a.png)

**System (please complete the following information):**
 - Operating System: Window 10
 - Java Version:
java version "14.0.2" 2020-07-14
Java(TM) SE Runtime Environment (build 14.0.2+12-46)
Java HotSpot(TM) 64-Bit Server VM (build 14.0.2+12-46, mixed mode, sharing)
 - Randomizer Version: 4.3.0

**Comment by tom-overton** [comment]
In Omega Ruby and Alpha Sapphire specifically, there's an absolute hard limit on the number of unique Pokemon species that can be on any given route. That number is specifically 18. If you exceed this limit, the game crashes as soon as you attempt to load the route.

As a consequence of this crashing behavior, for Omega Ruby and Alpha Sapphire only, our "totally random" setting actually isn't totally random; it's limiting itself to prevent a crash. What you're seeing in this log is the result of this limitation. We still try our best to add some degree of extra randomness while avoiding the crash, but we are quite limited. Some areas have dozens upon dozens on encounter slots, but we are only allowed 18 unique species, so duplicates are absolutely mandatory.

In short, there's no bug here; this behavior is entirely intentional to prevent the game from crashing. Outside of removing the hardcoded limit in the game itself, there's nothing more we can do.
